### PR TITLE
updated sdk to work with guzzle 6.0

### PR DESCRIPTION
**Pull request by leftyhitchens**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "type": "library",
     "require": {
         "php": ">=5.5.9",
-        "guzzlehttp/guzzle": "5.*"
+        "guzzlehttp/guzzle": "~6.0"
```
Comment (leftyhitchens) [inline composer.json]: Here added guzzle 6.0 as composer requirement

File: src/AvaTaxClient.php [hunk 2]
```diff
@@ -49,6 +59,12 @@ class AvaTaxClient
      */
     public function __construct($appName, $appVersion, $machineName, $environment)
     {
+
+        $this->appName = $appName;
```
Comment (leftyhitchens) [inline src/AvaTaxClient.php]: Adding these as instance variables so they can be used below in ```restCall```

File: src/AvaTaxClient.php [hunk 3]
```diff
@@ -59,13 +75,8 @@ public function __construct($appName, $appVersion, $machineName, $environment)
 
         // Configure the HTTP client
         $this->client = new Client([
-            'base_url' => $env
+            'base_uri' => $env
         ]);
-        
-        // Set client options
-        $this->client->setDefaultOption('headers', array(
```
Comment (leftyhitchens) [inline src/AvaTaxClient.php]: There is no longer a method for ```setDefaultOption``` in guzzle 6

File: src/AvaTaxClient.php [hunk 4]
```diff
@@ -5644,11 +5655,14 @@ private function restCall($apiUrl, $verb, $guzzleParams)
         if (!isset($guzzleParams['auth'])){
             $guzzleParams['auth'] = $this->auth;
         }
-    
+        $guzzleParams['headers'] = [
```
Comment (leftyhitchens) [inline src/AvaTaxClient.php]: Adding headers before each request

**Comment by leftyhitchens** [comment]
The build is failing but I think it is just because the environment variables aren't set on the travis ci build correctly.

**Comment by ted-spence-avalara** [comment]
@leftyhitchens  - Does upgrading to Guzzle 6 break compatibility with older PHP versions?  I am asking my team to do review on their platforms.

**Comment by leftyhitchens** [comment]
@ted-spence-avalara  according to the documentation on Guzzle's github page guzzle 6's minimum php version is 5.5.

https://github.com/guzzle/guzzle

**Comment by ted-spence-avalara** [comment]
@leftyhitchens I'm working on the July release patch now - let's get this merged in for July as well.

**Comment by leftyhitchens** [comment]
@nitinshirsat No problem, thanks for the merge

**Pull request merged**